src/fnmatch.c

```diff
@@ -119,15 +122,18 @@ p_fnmatchx(const char *pattern, const char *string, int flags, size_t recurs)
 								break;
 						}
 
+						bool string_path_delimiter_start = true;
 						/* General case, use recursion. */
 						while ((test = *string) != EOS) {
 								int e;
 
 								e = p_fnmatchx(pattern, string, recurs_flags, recurs);
-								if (e != FNM_NOMATCH)
-										return e;
-								if (test == '/' && (flags & FNM_PATHNAME))
-										break;
+								if (e != FNM_NOMATCH && !(pattern_path_delimiter_start && !string_path_delimiter_start)) {
```

Comment:
Thinking more about this, I guess we might also break early when a match is found but it doesn't satisfy the path delimiter constraint:

```c
if (e != FNM_NOMATCH) {
	/* Break if the string doesn't start with a path delimiter but the pattern does. */
	if (pattern_path_delimiter_start && !string_path_delimiter_start) {
		break;
	} else {
		return e;
	}
}
```

Thoughts? :thought_balloon: